lua/telescope/builtin/__lsp.lua

```diff
@@ -106,8 +106,27 @@ end
 local function item_to_location(item, offset_encoding)
   local line = item.lnum - 1
   local character = vim.lsp.util._str_utfindex_enc(item.text, item.col, offset_encoding) - 1
+  -- get URI from filename with the same approach used in `vim.uri_from_bufnr`
+  local URI_SCHEME_PATTERN = '^([a-zA-Z]+[a-zA-Z0-9.+-]*):.*'
+  local WINDOWS_URI_SCHEME_PATTERN = '^([a-zA-Z]+[a-zA-Z0-9.+-]*):[a-zA-Z]:.*'
+  local uri
+  local fname = item.filename
+  local volume_path = fname:match('^([a-zA-Z]:).*')
+  local is_windows = volume_path ~= nil
+  local scheme
+  if is_windows then
+    fname = fname:gsub('\\', '/')
+    scheme = fname:match(WINDOWS_URI_SCHEME_PATTERN)
+  else
+    scheme = fname:match(URI_SCHEME_PATTERN)
+  end
```

Comment:
we already have `utils.is_uri`

Reply:
Thank you @Conni2461, I did not know about `utils.is_uri`. I wish I knew. I have updated the logic accordingly.